gapis/api/vulkan/replay.go

```diff
@@ -783,13 +767,16 @@ func (a API) QueryFramebufferAttachment(
 	if err != nil {
 		return nil, err
 	}
+	if _, ok := mgr.(replay.Exporter); ok {
```

Comment:
This is nasty. Can't we just fake the POST data (say with 0's) in the exporter so the API's don't need to care about the manager implementation?

Reply:
It's not really possible as the exporter need to be aware of the type of the POST data. 
It's possible to ignore type check as
```
data, _ :=res.(*image.Data)
return data, nil
```
however it only hide the problem.

Reply:
What do you mean by type? I don't think there's anything in the postback handlers that will explode if the buffer returned contained unexpected (0) data. In the case it did check, it should just return an error once parsed, which will be ignored by your new manager anyway.

What I'm suggesting is:

Postbacks are requested by the API using the [`replay.Builder.Post()`](https://github.com/google/gapid/blob/master/gapis/replay/builder/builder.go#L474) function.
This [appends a `postBackDecoder`](https://github.com/google/gapid/blob/master/gapis/replay/builder/builder.go#L482) to the [`replay.Builder.decoders` slice](https://github.com/google/gapid/blob/master/gapis/replay/builder/builder.go#L89).
If you added a new method to `replay.Builder` to return the list of expected messages, you could call [the returned `PostDataHandler`](https://github.com/google/gapid/blob/master/gapis/replay/builder/builder.go#L634-L656) with 0 data for each of the expected postbacks.